plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -231,7 +242,7 @@ private static Object resolve(String pathElement, String fullPath, Object contex
      * @throws IllegalArgumentException if the path is null, empty, invalid or if the field doesn't exist.
      */
     public void appendFieldValue(String path, Object value) {
-        setFieldValue(path, value, true);
+        setFieldValue(TemplateService.wrap(path), ValueSource.wrap(value), true);
```

Comment:
ok but I think the templated methods should call the simple ones providing the result of the templates execution and not the other way around?